pkg/manifest/manifest_test.go

```diff
@@ -155,6 +161,76 @@ data:
 
 }
 
+func TestParseManifestsDuplicates(t *testing.T) {
+	tests := []struct {
+		name    string
+		raw     string
+		wantErr string
+	}{{
+		name: "no-duplicate",
+		raw: `
+apiVersion: extensions/v1
+kind: ConfigMap
+metadata:
+  name: a-config
+  namespace: default
+---
+apiVersion: extensions/v1
+kind: ConfigMap
+metadata:
+  name: b-config
+  namespace: default
+`,
+		wantErr: "",
+	}, {
+		name: "resource-id-error",
+		raw: `
+apiVersion: extensions/v1
+kind: Kind
+metadata:
+  name:
+  namespace: default
+`,
+		wantErr: "must contain kubernetes required fields kind and name",
+	}, {
+		name: "duplicate",
+		raw: `
+apiVersion: extensions/v1
+kind: ConfigMap
+metadata:
+  name: a-config
+  namespace: default
+---
+apiVersion: extensions/v2
+kind: ConfigMap
+metadata:
+  name: a-config
+  namespace: default
+`,
+		wantErr: "duplicate resource:",
+	}}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := ParseManifests(strings.NewReader(test.raw))
+			if err == nil {
+				if len(test.wantErr) != 0 {
+					t.Fatalf("Expected an error and got none")
+				}
+			} else {
+				if len(test.wantErr) == 0 {
```

Comment:
nit: you can drop one level of nesting with:

```go
if err == nil {
  ...your existing no-error handling...
} else if len(test.wantErr) == 0 {
  t.Fatalf("Got unexpected error: %v", err)
} else if !strings.Contains(err.Error(), test.wantErr) {
  t.Fatalf("Got incorrect error. Wanted it to contain %s but got %s", test.wantErr, err.Error())
}
```

Reply:
agree